expression/builtin_compare.go

```diff
@@ -1359,13 +1360,22 @@ func (c *compareFunctionClass) refineArgs(ctx sessionctx.Context, args []Express
 		return []Expression{NewOne(), NewZero()}
 	}
 
-	return c.refineArgsByUnsignedFlag(ctx, []Expression{finalArg0, finalArg1})
+	return c.refineArgsByTypeRange(ctx, []Expression{finalArg0, finalArg1})
 }
 
-func (c *compareFunctionClass) refineArgsByUnsignedFlag(ctx sessionctx.Context, args []Expression) []Expression {
-	// Only handle int cases, cause MySQL declares that `UNSIGNED` is deprecated for FLOAT, DOUBLE and DECIMAL types,
+// refineArgsByTypeRange handle the case that constant exceeds the range corresponding to the type of column when `column <cmp> constant`.
+// If necessary, we will convert the result of the comparison into constant true or constant false.
+func (c *compareFunctionClass) refineArgsByTypeRange(ctx sessionctx.Context, args []Expression) []Expression {
+	// We should guarantee the number of arguments are equal to two.
+	if len(args) != 2 {
```

Comment:
This is redundant since we have checked this restriction in `compareFunctionClass::getFunction()`?